lib/ddtrace/contrib/rack/rum_injection.rb

```diff
@@ -0,0 +1,239 @@
+            !streaming?(headers, env) &&
+            injectable_html?(headers)
+        # catch everything and swallow it here for defensiveness
+        rescue Exception => e # rubocop:disable Lint/RescueException
+          Datadog.logger.warn("Error during rum injection  #{e.class}: #{e.message} #{e.backtrace.join("\n")}")
+          return nil
+        end
+
+        def compressed?(headers)
+          headers && headers.key?(CONTENT_ENCODING_HEADER) &&
+            !headers[CONTENT_ENCODING_HEADER].start_with?(IDENTITY)
+        end
+
+        def injectable_html?(headers)
+          (headers && headers.key?(CONTENT_TYPE_HEADER) && !headers[CONTENT_TYPE_HEADER].nil?) &&
+            (headers[CONTENT_TYPE_HEADER].include?(HTML_CONTENT) ||
+            headers[CONTENT_TYPE_HEADER].include?(XHTML_CONTENT))
+        end
+
+        def attachment?(headers)
+          (headers && headers.key?(CONTENT_DISPOSITION_HEADER) && !headers[CONTENT_DISPOSITION_HEADER].nil?) &&
+            !headers[CONTENT_DISPOSITION_HEADER].include?(INLINE)
+        end
+
+        def streaming?(headers, env)
+          # https://api.rubyonrails.org/classes/ActionController/Streaming.html
+          # rails recommends disabling middlewares that interact with response body
+          # when streaming via ActionController::Streaming
+          # TODO: if required to patch streaming, investigate patching further upstream, in the render action perhaps
+          return true if (headers && headers.key?(TRANSFER_ENCODING_HEADER) &&
+                         headers[TRANSFER_ENCODING_HEADER] == TRANSFER_ENCODING_CHUNKED) ||
+                         (
+                            headers && headers.key?(CONTENT_TYPE_HEADER) &&
+                            !headers[CONTENT_TYPE_HEADER].nil? &&
+                            headers[CONTENT_TYPE_HEADER].start_with?(CONTENT_TYPE_STREAMING)
+                         )
+
+          # if we detect Server Side Event streaming controller, assume streaming
+          defined?(ActionController::Live) &&
+            env[ACTION_CONTROLLER_INSTANCE].class.included_modules.include?(ActionController::Live)
+        end
+
+        def no_cache?(headers, env)
+          # TODO: this is very complex, is there an easier way to determine cache behavior on cdn and browser
+          # TODO: glob performance may be worse than regex
+          return false unless configuration[:rum_cached_pages].none? do |page_glob|
+            File.fnmatch(page_glob, env['PATH_INFO']) if env['PATH_INFO']
+          end
+
+          return false unless headers
+
+          # first check Surrogate-Controle, which fastly uses
+          # indicates a cdn cache if Surrogate-Control max-age>0,
+          # otherwise check cache-control, then expiry to determine if there is browser cache
+          return false if surrogate_cache?(headers)
+
+          # then check Cache-Control
+          if headers.key?(CACHE_CONTROL_HEADER) && !headers[CACHE_CONTROL_HEADER].nil?
```

Comment:
```suggestion
          if cache_control_header = headers[CACHE_CONTROL_HEADER]
```

Let's also refer to `cache_control_header` instead of re-fetching it form the headers array again.

Reply:
👍 makes sense. Rubocop yells at me when i do assignment in a conditional so can't be _as_ DRY as i'd like, but cleaned this up to use a variable

Reply:
Rubocop won't yell if you put it inside parenthesis I believe (`if (cache_control_header = headers[CACHE_CONTROL_HEADER])`).